client/src/Authentication/LoginForm.js

```diff
@@ -95,14 +94,28 @@ class LoginForm extends React.Component {
                     </div>
                     {/* TODO: Clean Up: */}
                     <div className="user-prompt">
-                        
+
                         {/* <h2>New User?</h2>
                         <a href="#">Click here to register a new account.</a> */}
                     </div>
                 </div>
 
             </div>
         )
+    };
+
+    openGoogleLogin = () => {
+        window.open("http://localhost:5000/auth/google/", "Login",'height=800,width=500');
+    };
+
+    componentDidMount = () => {
+        window.addEventListener('message', (event) => {
```

Comment:
don't like this post back to the parent window, maybe consider a solution like this? https://stackoverflow.com/questions/28392393/passport-js-after-authentication-in-popup-window-close-it-and-redirect-the-pa

Reply:
I did try the method in the link. Right now the node.js server and React server are on different ports so I call to the main window's function will be blocked. The current solution is a workaround. In our final product, will the node server and the react server on the same domain name (via proxy or sth)?

Reply:
ah interesting, in the end yes, the express backend will actually serve a static react app. the current separation is for development (i.e. the dev tool create-react-app runs a separate dev server) but in production, we'll build and deploy react app via express. see this [article](https://daveceddia.com/deploy-react-express-app-heroku/)